homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -189,7 +202,9 @@ async def async_step_user(self, user_input=None):
 
         return self.async_create_entry(title=host, data=config_data)
 
-    async def async_step_ssdp(self, discovery_info):
+    async def async_step_ssdp(
+        self, discovery_info: dict[str, Any]
```

Comment:
`from homeassistant.helpers.typing import DiscoveryInfoType`
```suggestion
        self, discovery_info: DiscoveryInfoType
```